portal/src/model/themeAuthFlowV2.ts

```diff
@@ -264,3 +320,50 @@ export class StyleCSSVisitor<T> extends CssNodeVisitor {
     return this.styleGroup.getValue();
   }
 }
+
+export class CssAstVisitor {
+  private root: Root;
+  private rule: Rule;
+
+  constructor(ruleSelector: string) {
+    this.root = new Root();
+    this.rule = new Rule({
+      selector: ruleSelector,
+    });
+    this.root.append(this.rule);
+  }
+
+  visitStyleGroup<T>(styleGroup: StyleGroup<T>): void {
+    for (const style of Object.values(styleGroup.styles)) {
+      const s = style as AbstractStyle<T>;
+      s.acceptCssAstVisitor(this);
+    }
+  }
+
+  visitAlignItemsStyleProperty(styleProperty: AlignItemsStyleProperty): void {
+    this.visitorStyleProperty(styleProperty);
+  }
+
+  visitBorderRadiusStyleProperty(
+    styleProperty: BorderRadiusStyleProperty
+  ): void {
+    this.visitorStyleProperty(styleProperty);
+  }
+
+  visitColorStyleProperty(styleProperty: ColorStyleProperty): void {
+    this.visitorStyleProperty(styleProperty);
+  }
+
+  visitorStyleProperty<T>(styleProperty: StyleProperty<T>): void {
+    this.rule.append(
+      new Declaration({
+        prop: styleProperty.propertyName,
+        value: String(styleProperty.getCSSValue()),
+      })
+    );
+  }
+
+  getCSS(): Root {
+    return this.root;
+  }
+}
```

Comment:
Can we add some tests to demonstrate how this thing works, and to prove it is working as expected?